project/iterative_sorting.py

```diff
@@ -18,8 +18,21 @@ def selection_sort(arr):
 
 # TO-DO: implement the Insertion Sort function below
 def insertion_sort(arr):
```

Comment:
this looks great, Kam, but I think you can still optimize a bit. is there a need to append every element to an array? (line 26)
see if you can have this sort in place

Reply:
I don't like sorting in place generally, it forces me to mutate values and makes it less flexible. Tried to return new values when possible. I thought it was harmless as append is O(1), but it appears there is reallocation that takes place, so it is not so cheap as I thought.

I'll take a look at doing in in place!

Reply:
I agree, in place sorting can be a bit scary sometimes (particularly with repeated values)

let me know how it goes!

Reply:
I reimplemented it already in place, it was a good experience:
```
def insertion_sort(arr):
    if not arr:
        return arr
    for i in range(1, len(arr)):
        j = i - 1
        while j >= 0 and arr[i] < arr[j]:
            arr[i], arr[j] = arr[j], arr[i]
            i = j
            j = i - 1
    return arr
```